pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java

```diff
@@ -74,60 +76,115 @@
   }
 
   public static class ControllerPeriodicTasksConf {
+
     // frequency configs
+    @Deprecated
     public static final String RETENTION_MANAGER_FREQUENCY_IN_SECONDS = "controller.retention.frequencyInSeconds";
+    public static final String RETENTION_MANAGER_FREQUENCY_PERIOD = "controller.retention.frequencyPeriod";
     @Deprecated
     // The ValidationManager has been split up into 3 separate tasks, each having their own frequency config settings
     public static final String DEPRECATED_VALIDATION_MANAGER_FREQUENCY_IN_SECONDS =
         "controller.validation.frequencyInSeconds";
+    @Deprecated
     public static final String OFFLINE_SEGMENT_INTERVAL_CHECKER_FREQUENCY_IN_SECONDS =
         "controller.offline.segment.interval.checker.frequencyInSeconds";
+    public static final String OFFLINE_SEGMENT_INTERVAL_CHECKER_FREQUENCY_PERIOD =
+        "controller.offline.segment.interval.checker.frequencyPeriod";
+    @Deprecated
     public static final String REALTIME_SEGMENT_VALIDATION_FREQUENCY_IN_SECONDS =
         "controller.realtime.segment.validation.frequencyInSeconds";
+    public static final String REALTIME_SEGMENT_VALIDATION_FREQUENCY_PERIOD =
+        "controller.realtime.segment.validation.frequencyPeriod";
+    @Deprecated
     public static final String REALTIME_SEGMENT_VALIDATION_INITIAL_DELAY_IN_SECONDS =
         "controller.realtime.segment.validation.initialDelayInSeconds";
+    public static final String REALTIME_SEGMENT_VALIDATION_INITIAL_DELAY_PERIOD =
+        "controller.realtime.segment.validation.initialDelayPeriod";
+    @Deprecated
     public static final String BROKER_RESOURCE_VALIDATION_FREQUENCY_IN_SECONDS =
         "controller.broker.resource.validation.frequencyInSeconds";
+    public static final String BROKER_RESOURCE_VALIDATION_FREQUENCY_PERIOD =
+        "controller.broker.resource.validation.frequencyPeriod";
+    @Deprecated
     public static final String BROKER_RESOURCE_VALIDATION_INITIAL_DELAY_IN_SECONDS =
         "controller.broker.resource.validation.initialDelayInSeconds";
+    public static final String BROKER_RESOURCE_VALIDATION_INITIAL_DELAY_PERIOD =
+        "controller.broker.resource.validation.initialDelayPeriod";
+    @Deprecated
     public static final String STATUS_CHECKER_FREQUENCY_IN_SECONDS = "controller.statuschecker.frequencyInSeconds";
+    public static final String STATUS_CHECKER_FREQUENCY_PERIOD = "controller.statuschecker.frequencyPeriod";
+    @Deprecated
     public static final String STATUS_CHECKER_WAIT_FOR_PUSH_TIME_IN_SECONDS =
         "controller.statuschecker.waitForPushTimeInSeconds";
+    public static final String STATUS_CHECKER_WAIT_FOR_PUSH_TIME_PERIOD =
```

Comment:
This config is (and please add this doc here) to make sure that a recently pushed segment is not accidentally flagged as bad just because it does not reflect in external view as yet. So, this time should be in seconds all the time. Please revert this change.

Reply:
@mcvsubbu this config's default value in [docs](https://docs.pinot.apache.org/configuration-reference/controller#:~:text=controller.statuschecker.waitForPushTimeInSeconds) is 10 minutes. So can't we expect that the user might want to supply a different value in minutes again?

Reply:
You are right. I was mixed up with another config perhaps. Thanks for the diligence